### PR TITLE
Remove generation field of instance

### PR DESCRIPTION
**Pull request by FireDefend**
following the latest spec, "generation " is an unnecessary filed. we will use the internal "generation" of K8S CR.

Removed "generation" from k8s instance spec, move the generation from api instance spec to ObjectMeta, Keep the Etag for StateEntry for the commit function when update StateEntry.

### CODE REVIEW

**Comment by LeiQL** [comment]
only instance `generation` needs to be removed?

**Review by Haishi2016 — COMMENTED** [review]
We may need to consider this. As we get rid of Instance (and Target) versioning, we may need to keep generations so that when user edits the instance object new reconciliation can be triggered.

File: api/pkg/apis/v1alpha1/model/instance.go [hunk 1]
```diff
@@ -31,7 +31,6 @@ type (
 		Target       TargetSelector    `json:"target,omitempty"`
 		Topologies   []TopologySpec    `json:"topologies,omitempty"`
 		Pipelines    []PipelineSpec    `json:"pipelines,omitempty"`
-		Generation   string            `json:"generation,omitempty"`
 		Version      string            `json:"version,omitempty"`
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/model/instance.go:34]: Can we also look at how this "Version" is used? This was introduced by AIO, not sure what its functionality is.

Comment (msftcoderdjw) [reply api/pkg/apis/v1alpha1/model/instance.go:34]: This seems to be introduced by @LeiQL for all versioned objects (version + rootResource property). Can you please confirm it?

**Comment by FireDefend** [comment]
Removed "generation" from k8s instance spec, move the generation from api instance spec to ObjectMeta, Keep the Etag for StateEntry for the commit function when update StateEntry.

**Review by msftcoderdjw — APPROVED** [review]

File: api/pkg/apis/v1alpha1/model/instance.go (hunk 1 above)
Comment (LeiQL) [reply api/pkg/apis/v1alpha1/model/instance.go:34]: Correct. Being remove in PR #323.

**Pull request merged**